internal-services/catalog/create-advisory-task.yaml

```diff
@@ -0,0 +1,111 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: create-advisory-task
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+      Pushes an advisory yaml to a Git repository
+  params:
+    - name: advisory_json
+      type: string
+      description: |
+          String containing a JSON representation of the advisory data. It should not contain the `advisory`
+          top level key (e.g. '{"spec":{"product_id":123,"type":"RHSA"}}')
+    - name: application
+      type: string
+      description: Application being released
+    - name: origin
+      type: string
+      description: |
+          The origin workspace where the release CR comes from.
+          This is used to determine the advisory path
+  results:
+    - name: result
+      description: Success if the task succeeds, the error otherwise
+  steps:
+    - name: create-advisory
+      image: quay.io/hacbs-release/release-utils:518c0423a3b3502c7b9ef6fd8652995bec2b4e1a
+      env:
+        - name: GITLAB_HOST
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_host
+        - name: ACCESS_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_access_token
+        - name: GIT_AUTHOR_NAME
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_name
+        - name: GIT_AUTHOR_EMAIL
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_email
+      script: |
+          #!/usr/bin/env sh
+          set -ox pipefail
+
+          # This should get overwritten if any command fails
+          echo -n "Success" > $(results.result.path)
+
+          failfunc() {
+              local err=$1
+              local line=$2
+              local command="$3"
+              echo "$0: ERROR '$command' failed at line $line - exited with status $err" > $(results.result.path)
+              exit 0 # exit the script cleanly as there is no point in proceeding past an error
```

Comment:
> as there is no point in proceeding past an error

Also below:

> the task should never fail with nonzero exit code

But why? I know the answer - because the internal request is called from the main RHTAP cluster and we want to be able to see the result there. Can you maybe explain that briefly somewhere? Either in one of the comments, or in the task description - that it will return 0 even if something fails and why.

Reply:
Looks good, thanks!